numpyro/distributions/continuous.py

```diff
@@ -246,6 +246,13 @@ def variance(self):
         return jnp.broadcast_to(jnp.expand_dims(self.scale, -1) ** 2 * jnp.arange(1, self.num_steps + 1),
                                 self.batch_shape + self.event_shape)
 
+    def tree_flatten(self):
+        return (self.scale,), self.num_steps
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, params):
+        return cls(*params, num_steps=aux_data)
```

Comment:
IIUC, `aux_data` is all the arguments to the distribution class that are not `DeviceArray`?

Reply:
You are right, those arguments are constant during `scan`.